### PR TITLE
Conversion Hosts: list view phase 1

### PR DESCRIPTION
**Pull request by mzazrivec**
This PR implements:
- basic list view for conversion hosts
- pagination
- filtering
- sorting

Currently, the list view displays only conversion host name. The issue linked below also mentions showing provider type (or perhaps provider name): for that we'd need more information to be pulled from API.

conversion-hosts-01

Fixes: https://github.com/ManageIQ/manageiq-v2v/issues/860

### CODE REVIEW

File: app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js [hunk 1]
```diff
@@ -1,17 +1,61 @@
...
+    }) => (
+      <React.Fragment>
+        <Grid.Row>
+          <Toolbar>
+            {renderFilterControls()}
+            {renderSortControls()}
+            {renderActiveFilters(filteredSortedPaginatedListItems)}
+          </Toolbar>
+        </Grid.Row>
+        <div style={{ overflow: 'auto', paddingBottom: 300, height: '100%' }}>
```
Comment (mturley) [inline app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js]: This is outside the scope of this PR, because this pattern is all over the place, but I think at some point we should probably move all of these inline `style` props into CSS, it's a little messy that we use both. It's fine for now though.

File: app/javascript/react/screens/App/Settings/Settings.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .migration-settings h2 {
   margin-bottom: 25px;
 }
+
+.conversion-hosts-list .list-view-pf-main-info {
```
Comment (mturley) [inline app/javascript/react/screens/App/Settings/Settings.scss]: This smaller padding is fine for now, but as we add more stuff to the list view items we'll probably want to remove or tweak this to make sure the layout matches the mockups and the patternfly standards. I think maybe once we have more in here, the thicker padding will make more sense.

File: app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js [hunk 3]
```diff
@@ -1,17 +1,61 @@
...
+          <ListView className="conversion-hosts-list" id="conversion_hosts">
+            {filteredSortedPaginatedListItems.items.map((host, n) => (
+              <ListView.Item key={host.id} heading={host.name} stacked />
```
Comment (mturley) [inline app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js]: Once we add more details to the list view items, they should probably become their own `ConversionHostListItem` component that we import here.

File: app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js [hunk 4]
```diff
@@ -1,17 +1,61 @@
...
+      </React.Fragment>
+    )}
+  </ListViewToolbar>
 );
 
 ConversionHostsList.propTypes = {
   conversionHosts: PropTypes.arrayOf(PropTypes.object)
 };
 
+ConversionHostsList.sortFields = [
```
Comment (mturley) [inline app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js]: I was about to comment that we should move these `sortFields` and `filterTypes` into a constants file, but I actually like them here as a property on the component. Unless we end up needing to reuse them somewhere else, this is a fine place for them.

File: app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js (hunk 1 above)
Comment (mzazrivec) [reply app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js]: Yep, I definitelly agree with moving this into CSS (we use it in 3 other places as well).

File: app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js (hunk 4 above)
Comment (mzazrivec) [reply app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostsList.js]: Yes, we actually use both approaches in v2v code, I'm fine with going either way with this, just perhaps use same pattern everywhere. What do you think?

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 96dcc9ebe2c93272cd82d7bc831dfab17c11beb4
Author: Mike Turley <mike.turley@alum.cs.umass.edu>
Date:   Tue Feb 5 18:33:44 2019 -0500

    Merge pull request #874 from mzazrivec/conversion_hosts_list_view_phase_1
    
    Conversion Hosts: list view phase 1
    
    (cherry picked from commit e379015fc4764ce0ab70e323313115e30443df67)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1696423
```
